pages/volunteer/volunteer.html

```diff
@@ -224,29 +75,29 @@ <h3 class="md-display-1">
         <md-button class="oppia-foundation-button-horizontal-padding oppia-foundation-gold-border-button volunteer-page-no-margin">Apply now</md-button>
       </md-content>
     </md-card-content>
-    <div show-xs show-sm hide-gt-sm>
-      <img src="/static/images/volunteer_design_profile_image.png" class="volunteer-page-design-profile-image">
-    </div>
-    <div show-xs show-sm hide-gt-sm class="oppia-foundation-quote-box-overlay volunteer-page-conditional-horizontal-padding">
-      <md-card-content layout="column" layout-align="center none" class="oppia-foundation-quote-box oppia-foundation-white-background">
-        <md-content layout="column" layout-align="center none" class="oppia-foundation-white-background oppia-foundation-white-background oppia-foundation-black-text">
-          <blockquote class="oppia-foundation-blockquote">
-            <span>&ldquo;</span>
-            As a former teacher, I am motivated to create a platform that gives everyone equal access to education.
-            Working with Oppia is a powerful opportunity to leverage design for social justice.
-            <span>&rdquo;</span>
-          </blockquote>
-        </md-content>
-        <cite class="oppia-foundation-cite-text-box">
-          <span class="oppia-foundation-bold-text">- Jordan Movish</span>
-          <p class="oppia-foundation-no-margin"></p>
-          <span class="oppia-foundation-italic-text">UX Design Lead</span>
-        </cite>
-      </md-card-content>
-    </div>
-
-    <md-card-content show-xs show-sm hide-gt-sm>
-      <md-content class="oppia-foundation-gray-background oppia-foundation-black-text">
+    <uib-carousel uib-carousel active="designSlide.active" interval="slideInterval" no-wrap="noWrapSlides">
```

Comment:
It would be cleaner if you could refactor this into a reusable directive, since exactly the same logic is being repeated four times. That would make the code easier to maintain.

Reply:
These uib-carousel relies on different slides. Is it possible to call a template with an argument?
Can you point me to a general doc?

Reply:
The Desktop 's carousel is also different from that of the Mobile's carousel (slightly). They are categorized by tab's type. Should I do the same here...i.e mobile_design_tab_template.html?

Reply:
Is [this SO post and its fiddle](https://stackoverflow.com/questions/26409460/angularjs-pass-argument-to-directive) what you are looking for.

I will attempt to mirror something like that to reduce duplicated HTML codes.

Reply:
Yes that's right. We do a lot of that sort of thing in the main oppia codebase (custom directives with attributes). You could implement a `<volunteer-page-carousel>` directive that wraps uib-carousel, for example.

If the templates are not too different, you could have the branching within the carousel directive (use some JS to determine the viewport width or the device type, say, and use ng-if to control what is shown). Otherwise you could have a separate template that handles the mobile case, and do the device type detection + ng-if branching in this file instead.

Reply:
Replaced duplicated carousel HTML codes with directive!